sw/device/lib/drivers/fast_intr_ctrl/fast_intr_ctrl.c

```diff
@@ -1,24 +1,431 @@
-// Copyright 2022 OpenHW Group
-// Solderpad Hardware License, Version 2.1, see LICENSE.md for details.
-// SPDX-License-Identifier: Apache-2.0 WITH SHL-2.1
+/*
+                              *******************
+******************************* C SOURCE FILE *****************************
+**                            *******************                          
+**                                                                         
+** project  : X-HEEP                                                       
+** filename : fast_intr_ctrl.c                                                      
+** version  : 1                                                            
+** date     : 27/03/23                                                     
+**                                                                         
+***************************************************************************
+**                                                                         
+** Copyright (c) EPFL contributors.                                        
+** All rights reserved.                                                    
+**                                                                         
+***************************************************************************
+*/
 
-#include "fast_intr_ctrl.h"
+/***************************************************************************/
+/***************************************************************************/
+/**
+* @file   fast_intr_ctrl.c
+* @date   27/03/23
+* @brief  The fast interrupt controller peripheral driver 
+*/
+
+/****************************************************************************/
+/**                                                                        **/
+/*                             MODULES USED                                 */
+/**                                                                        **/
+/****************************************************************************/
 
+#include "fast_intr_ctrl.h"
 #include <stddef.h>
 #include <stdint.h>
-
 #include "mmio.h"
-
 #include "core_v_mini_mcu.h"
-
 #include "fast_intr_ctrl_regs.h"  // Generated.
+#include "fast_intr_ctrl_structs.h"
+
+/* interrupts */
+// #include "handler.h"
+#include "csr.h"
```

Comment:
why do you need to include this?

Reply:
you are right, removed, it was there from pre version